assets/js/modules/analytics/components/settings/SettingsForm.js

```diff
@@ -31,39 +36,78 @@ import {
 	UseSnippetSwitch,
 	ProfileNameTextField,
 	ExistingGTMPropertyNotice,
+	GA4Notice,
+	GA4PropertyNotice,
 } from '../common';
+import GA4PropertySelect from '../../../analytics-4/components/common/PropertySelect';
 import StoreErrorNotices from '../../../../components/StoreErrorNotices';
-import GA4Notice from '../common/GA4Notice';
-import { STORE_NAME } from '../../datastore/constants';
+import {
+	SETUP_FLOW_MODE_GA4,
+	SETUP_FLOW_MODE_GA4_TRANSITIONAL,
+	SETUP_FLOW_MODE_LEGACY,
+	SETUP_FLOW_MODE_UA,
+	STORE_NAME,
+	PROFILE_CREATE,
+} from '../../datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import { useFeature } from '../../../../hooks/useFeature';
 const { useSelect } = Data;
 
 export default function SettingsForm() {
+	const isGA4Enabled = useFeature( 'ga4setup' );
+	useSelect( ( select ) => {
+		if ( ! isGA4Enabled ) {
+			return;
+		}
+
+		// We need to call getProperties for getSetupFlowMode to work.
+		const accountID = select( STORE_NAME ).getAccountID();
+		return select( MODULES_ANALYTICS_4 ).getProperties( accountID );
```

Comment:
This seems a bit of a hack—relying on the side effect of a selector to call another one.

It makes me think we should be calling these selectors as part of `getSetupFlowMode`'s resolver.

Why can't we do that here?

Reply:
My initial idea was to change that resolver, however it wasn't in the IB, otherwise this hack has to happen everywhere we call the selector.

Reply:
As in: if we don't change the resolver this needs to be added everywhere we call the `getSetupFlowMode`'s resolver?

That's a recipe for bugs in my mind; I see it isn't in the IB, but let's modify the resolver/selector to call these dependant selectors first.

----

Actually, I'm a bit confused; I went to look at the selector for `getSetupFlowMode` and it already calls the selectors you're looking for: https://github.com/google/site-kit-wp/blob/922999e8bb12bb40108244e992b61e5c9c07f88f/assets/js/modules/analytics/datastore/setup-flow.js#L56-L74

Reply:
> Actually, I'm a bit confused; I went to look at the selector for getSetupFlowMode and it already calls the selectors you're looking for:

The `getProperties` may return undefined because it has to be resolved first. That's why we call it beforehand.

Reply:
Once it's resolved the selectors should all update though, no?

We account for that here: https://github.com/google/site-kit-wp/blob/b252720322e3dce7883c92c79811c6d83e19f20c/assets/js/modules/analytics/datastore/setup-flow.js#L76-L78

If `undefined` is returned by `getSetupFlowMode` we should render a loading screen/null/something else until things have loaded. The selector is designed to work without needing things to resolve first, I think here we just have to account for it returning `undefined` while things load.

Reply:
`isAdminAPIWorking()` returns undefined without it
https://github.com/google/site-kit-wp/blob/922999e8bb12bb40108244e992b61e5c9c07f88f/assets/js/modules/analytics-4/datastore/api.js#L32

which gets us to here:
https://github.com/google/site-kit-wp/blob/b252720322e3dce7883c92c79811c6d83e19f20c/assets/js/modules/analytics/datastore/setup-flow.js#L49

Reply:
@tofumatt let's create a follow-up ticket for this if you think it's worth it and not block this PR because it is not relevant to this ticket.